File: front/src/app/Domain/Models/IPedido.ts
import { IEmpresa } from './IEmpresa';
import { IPedidoItem } from './IPedidoItem';
import { ITcsprj } from './ITcsprj';
import { ITgfpar } from './ITgfpar';
import { ITgftpv } from './ITgftpv';
import { IUsuario } from './IUsuario';

export interface Pedido {
  id: number;
  frete: string;
  clienteCod: number;
  comprador: string;
  observacao: string;
  remessa: boolean;
  clienteRemCod: number;
  condPagCodTipVenda: number;
  condPagDhAlter: Date;
  status: string;
  dtCriacao: Date;
  dtEnvio: Date;
  pedSankhyaNuNota: number;
  orcamento: boolean;
  usuarioId: number;
  empresaId: number;
  estoque: string;
  pilotagem: boolean;
  tipoPilotagem: number;
  projetoCod: number;
  vendedorPCod: number;
  regraCif: string;
  ligarAntes: string;
  redpTel: string;
  redpRazao: string;
  redpCnpj: string;
  redp: string;
  ld: string;
  mediaNeg: number;
  confirmado: string;
  msgConfirmado: string;
  cnpjNovoCliente: string;
  tipoFrete: string;
  contatoCod: number;
  contatoCodParc: number;
  redespachoCodRed: number;
  dtBaseFin: Date;
  dtFat: Date;
  ordemCompra: string;
  pedidoItemDtFat: boolean;
  pedidoItemOrdemComp: boolean;
  diasVenc: number;
  dtCartao: Date;
  empresa: IEmpresa;
  usuario: IUsuario;
  tgftpv: ITgftpv;
  tcsprj: ITcsprj;
  tgfpar: ITgfpar;
  pedidoItem: IPedidoItem[];
}

File: back/back/domain/entities/IPedido.cs
using System;

namespace back.domain.entities
{
    public interface IPedido
    {
        public int Id { get; set; }
        public string Frete { get; set; }
        public Nullable<int> ClienteCod { get; set; }
        public string Comprador { get; set; }
        public string Observacao { get; set; }
        public bool Remessa { get; set; }
        public Nullable<int> ClienteRemCod { get; set; }
        public Nullable<int> CondPagCodTipVenda { get; set; }
        public Nullable<DateTime> CondPagDhAlter { get; set; }
        public string Status { get; set; }
        public Nullable<DateTime> DtCriacao { get; set; }
        public Nullable<DateTime> DtEnvio { get; set; }
        public Nullable<int> PedSankhyaNuNota { get; set; }
        public bool Orcamento { get; set; }
        public int UsuarioId { get; set; }
        public int EmpresaId { get; set; }
        public string Estoque { get; set; }
        public bool Pilotagem { get; set; }
        public Nullable<Int16> TipoPilotagem { get; set; }
        public Nullable<int> ProjetoCod { get; set; }
        public Nullable<int> VendedorPCod { get; set; }
        public Nullable<char> RegraCif { get; set; }
        public Nullable<char> LigarAntes { get; set; }
        public string RedpTel { get; set; }
        public string RedpRazao { get; set; }
        public string RedpCnpj { get; set; }
        public Nullable<char> Redp { get; set; }
        public Nullable<char> LD { get; set; }
        public Nullable<Int16> MediaNeg { get; set; }
        public Nullable<char> Confirmado { get; set; }
        public string MsgConfirmado { get; set; }
        public string CnpjNovoCliente { get; set; }
        public Nullable<char> TipoFrete { get; set; }
        public Nullable<int> ContatoCod { get; set; }
        public Nullable<int> ContatoCodParc { get; set; }
        public Nullable<int> RedespachoCodRed { get; set; }
        public Nullable<DateTime> DtBaseFin { get; set; }
        public Nullable<DateTime> DtFat { get; set; }
        public string OrdemCompra { get; set; }
        public Nullable<bool> PedidoItemDtFat { get; set; }
        public Nullable<bool> PedidoItemOrdemComp { get; set; }
        public Nullable<int> DiasVenc { get; set; }
        public Nullable<DateTime> DtCartao { get; set; }

    }
}
